requests/sessions.py

```diff
@@ -110,7 +110,7 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
                 resp.raw.read(decode_content=False)
 
             if i >= self.max_redirects:
-                raise TooManyRedirects('Exceeded %s redirects.' % self.max_redirects)
+                raise TooManyRedirects('Exceeded %s redirects.' % self.max_redirects, response=resp)
```

Comment:
Switched to just setting the response.  `RequestException` does `self.request = self.response.request`, so I figured that would be more consistent.